CHANGELOG.md

```diff
@@ -1,331 +1,1350 @@
+* If you're manually parsing multiaddrs and are searching for the string `/ipfs/`..., you'll need to search for `/p2p/...`.
+
+##### Minimum RSA Key Size
+
+Previously, IPFS did not enforce a minimum RSA key size. In this release, we've introduced a minimum 2048 bit RSA key size. IPFS generates 2048 bit RSA keys by default so this shouldn't be an issue for anyone in practice. However, users who explicitly chose a smaller key size will not be able to communicate with new nodes.
+
+Unfortunately, the some of the bootstrap peers _did_ intentionally generate 1024 bit RSA keys so they'd have vanity peer addresses (starting with QmSoL for "solar net"). All IPFS nodes should _also_ have peers with >= 2048 bit RSA keys in their bootstrap list, but we've introduced a migration to ensure this.
+
+We implemented this change to follow security best practices and to remove a potential foot-gun. However, in practice, the security impact of allowing insecure RSA keys should have been next to none because IPFS doesn't trust other peers on the network anyways.
+
+##### TLS By Default
+
+In this release, we're switching TLS to be the _default_ transport. This means we'll try to encrypt the connection with TLS before re-trying with SECIO.
+
+Contrary to the announcement in the go-ipfs 0.4.23 release notes, this release does not remove SECIO support to maintain compatibility with js-ipfs.
+
+Note: The `Experimental.PreferTLS` configuration option is now ignored.
+
+##### SECIO Deprecation Notice
 
 SECIO should be considered to be well on the way to deprecation and will be
 completely disabled in either the next release (0.6.0, ~mid May) or the one
 following that (0.7.0, ~end of June). Before SECIO is disabled, support will be
 added for the NOISE transport for compatibility with other IPFS implementations.
 
-#### QUIC Upgrade
+##### QUIC Upgrade
+
+If you've been using the experimental QUIC support, this release upgrades to a new and _incompatible_ version of the QUIC protocol (draft 27). Old and new go-ipfs nodes will still interoperate, but not over the QUIC transport.
+
+We intend to standardize on this draft of the QUIC protocol and enable QUIC by default in the next release if all goes well.
+
+NOTE: QUIC does not yet support [private networks](./docs/experimental-features.md#private-networks).
+
+#### Gateway
+
+In addition to a bunch of bug fixes, we've made two improvements to the gateway.
+
+You can play with both of these features by visiting:
+
+> http://bafybeia6po64b6tfqq73lckadrhpihg2oubaxgqaoushquhcek46y3zumm.ipfs.io/
+
+##### Subdomain Gateway
+
+First up, we've changed how URLs in the IPFS gateway work for better browser
+security. The gateway will now redirect from
+`http://localhost:5001/ipfs/CID/...` to `http://CID.ipfs.localhost:5001/...` by
+default. This:
+
+* Ensures that every dapp gets its own browser origin.
+* Makes it easier to write websites that "just work" with IPFS because absolute paths will now work.
```

Comment:
```suggestion
* Makes it easier to write websites that "just work" with IPFS because absolute paths will now work (though you should still use relative links because they're better).
```